src/main/kotlin/org/wycliffeassociates/otter/common/domain/resourcecontainer/project/usfm/UsfmProjectReader.kt

```diff
@@ -33,65 +48,87 @@ class UsfmProjectReader: IProjectReader {
         return Pair(result, projectTree)
     }
 
-    private fun parseFileIntoProjectTree(file: File, root: Tree, projectIdentifier: String): ImportResult {
-        return when (file.extension) {
-            "usfm", "USFM" -> {
-                try {
-                    val chapters = parseUSFMToChapterTrees(file, projectIdentifier)
-                    root.addAll(chapters)
-                    ImportResult.SUCCESS
-                } catch (e: RuntimeException) {
-                    ImportResult.INVALID_CONTENT
+    private fun constructTreeFromZip(container: ResourceContainer, project: Project): Pair<ImportResult, Tree> {
+        if (!project.path.endsWith(".usfm", ignoreCase = true)) {
+            return Pair(ImportResult.UNSUPPORTED_CONTENT, Tree(Unit))
+        }
+        // Find the appropriate zip entry and use it to construct the project tree
+        ZipFile(container.file).use { zip ->
+            val projectPath = currentDirectoryPrefix.replace(project.path, "")
+            zip.getEntry(projectPath)?.let {
```

Comment:
I think there's some risk that the zip file won't find the entry because it uses Windows separators, but let's defer figuring that out.